api/src/main/java/org/apache/iceberg/expressions/Literals.java

```diff
@@ -419,6 +420,9 @@ static class TimeLiteral extends ComparableLiteral<Long> {
     public <T> Literal<T> to(Type type) {
       if (type.typeId() == Type.TypeID.TIME) {
         return (Literal<T>) this;
+      } else if (type.typeId() == Type.TypeID.STRING) {
+        return (Literal<T>) new StringLiteral(LocalTime.ofNanoOfDay(value() * 1000)
+            .format(DateTimeFormatter.ISO_LOCAL_TIME));
```

Comment:
I don't think this conversion should be done using literals. Instead, use the methods in `DateTimeUtil` and add any new methods there. This shouldn't require changes to the public API in expressions.

Reply:
Yeah since previously I remember you pointed me to the `Literals` code so I thought I might be able to utilize the existing literals conversion logic there. 
My changes basically add the conversion from the 3 time/date-related type to string literal. Since I noticed the existing string literal code has the conversion to those date/time types, so I thought it might be a plus to add my current changes that can make the conversions more symmetric and enable people to do round-trip conversions?

WDYT?

Reply:
I don't think this conversion should be done using literals. You can see how we're doing it here, but there is no need to change the `Literals` API